packages/react-components/src/components/chart/chartOptions/useChartConfiguration.ts

```diff
@@ -0,0 +1,183 @@
+import {
+  DataPoint,
+  DataStream,
+  Threshold,
+  ThresholdValue,
+} from '@iot-app-kit/core';
+import {
+  ChartRef,
+  useGroupableEChart,
+  useLoadableEChart,
+} from '../../../hooks/useECharts';
+import { ChartOptions } from '../types';
+import { useXAxis } from './axes/xAxis';
+import { useTooltip } from './convertTooltip';
+import { useEffect, useMemo, useRef, useState } from 'react';
+import isEqual from 'lodash.isequal';
+import {
+  DEFAULT_CHART_OPTION,
+  PERFORMANCE_MODE_THRESHOLD,
+} from '../eChartsConstants';
+import { useSeriesAndYAxis } from './seriesAndYAxis/convertSeriesAndYAxis';
+
+const useTitle = ({
+  titleText,
+  hasSeries,
+}: {
+  titleText?: string;
+  hasSeries: boolean;
+}) => {
+  return useMemo(
+    () => ({
+      text: hasSeries ? titleText ?? '' : 'No data present',
+    }),
+    [hasSeries, titleText]
+  );
+};
+
+const toDataStreamIdentifiers = (dataStreams: DataStream[]) =>
+  dataStreams.map(({ id, name, color, refId }) => ({ id, name, color, refId }));
+
+type ChartConfigurationOptions = Pick<
```

Comment:
nit: probably worth cleaning this up a bit or just creating a new type object

Reply:
Can we address nits in a separate PR? I would like to get this into rc soon so that it can be validated by everyone working on features.